### PR TITLE
Added support for usergroups & services

### PR DESCRIPTION
**Pull request by michaelfmnk**
https://github.com/OpenVPN/terraform-provider-openvpn-cloud/issues/37

### CODE REVIEW

File: openvpncloud/resource_service.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+						},
+					},
+				},
+			},
+			"service_types": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
```
Comment (patoarvizu) [inline openvpncloud/resource_service.go:114]: The `validation` library has a built-in function for checking that the value is in a string slice https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/validation/strings.go#LL132C67-L132C67 (unless you want to have a very specific error message).

File: openvpncloud/resource_service.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type: schema.TypeString,
+				ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
```
Comment (patoarvizu) [inline openvpncloud/resource_service.go]: Same here, you can use the `StringLenBetween` validation function https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/validation/strings.go#L71

File: openvpncloud/resource_service.go (hunk 1 above)
Comment (patoarvizu) [reply openvpncloud/resource_service.go:114]: You can use that same function in the other fields where you're explicitly checking for values

Comment (michaelfmnk) [reply openvpncloud/resource_service.go:114]: thanks! replaced with `validation.StringInSlice`

File: openvpncloud/resource_service.go (hunk 2 above)
Comment (michaelfmnk) [reply openvpncloud/resource_service.go]: fixed this

**Review by patoarvizu — APPROVED** [review]
👍

**Review by arslanbekov — APPROVED** [review]
LGTM

**Pull request merged**